src/thor/bidirectional_astar.cc

```diff
@@ -1464,5 +1490,54 @@ bool IsBridgingEdgeRestricted(GraphReader& graphreader,
   return false;
 }
 
+TimeInfo EstimateReverseStartTime(GraphReader& reader,
+                                  valhalla::Location& origin,
+                                  valhalla::Location& destination,
+                                  const double factor,
+                                  const TimeInfo& time_info,
+                                  const sif::cost_ptr_t& costing,
+                                  const bool arrive_by) {
+  // correlations are sorted by distance
+  // so we just take the beeline distance between the
+  // origin and destination's first correlated points
+  auto origin_pathedge = origin.correlation().edges().at(0);
+  auto dest_pathedge = destination.correlation().edges().at(0);
+  PointLL origin_pt{origin_pathedge.ll().lng(), origin_pathedge.ll().lat()};
+  PointLL destination_pt{dest_pathedge.ll().lng(), dest_pathedge.ll().lat()};
+  auto dist = origin_pt.Distance(destination_pt);
+  auto seconds = costing->BeeLineTimeEstimate(dist, factor);
+  LOG_DEBUG("Estimated seconds: " + std::to_string(seconds));
+
+  graph_tile_ptr tile = nullptr;
+  valhalla::PathEdge& edge = arrive_by ? origin_pathedge : dest_pathedge;
+
+  // we need to get the node info either from the edge's start or end node
+  // for the timezone info
+  GraphId edgeid(edge.graph_id());
+  tile = reader.GetGraphTile(edgeid);
+
+  if (tile == nullptr) {
+    return TimeInfo::invalid();
+  }
+
+  const DirectedEdge* directededge = tile->directededge(edgeid);
+  graph_tile_ptr endtile = reader.GetGraphTile(directededge->endnode());
+
+  if (!endtile) {
+    return TimeInfo::invalid();
+  }
+  const NodeInfo* nodeinfo = nullptr;
+  if (arrive_by) {
+    nodeinfo = endtile->node(directededge->endnode());
+    return time_info.forward(seconds, static_cast<int>(nodeinfo->timezone()));
+  } else {
+    // in case of depart_at, we want the start node of our correlated edge
```

Comment:
Not sure we need differentiate which edge we take, unless we do better guessing based on the `percent_along`. TBH I'm ok with leaving a comment and simply take the first or last node no matter if fwd/rev